### ISSUE TITLE
Blaze client config should have a max wait on a connection

### ISSUE THREAD

**Issue by rossabaker** [http4s/http4s#1042]
* If it takes more than N seconds, fail.
* Don't leak a connection if it's created after we fail.

**Comment by rossabaker** [comment]
cats-effect-0.10 can probably help here.

**Comment by rossabaker** [comment]
Duplicate of #836
